src/ObjectSerializer.php

```diff
@@ -282,8 +282,13 @@ public static function deserialize($data, $class, $httpHeaders = null)
             }
 
             $file = fopen($filename, 'w');
-            while ($chunk = $data->read(200)) {
-                fwrite($file, $chunk);
+            if(is_string($data)){
```

Comment:
I think it would be preferred to also save the result in a variable and as  an alternative save the file. You could achieve backwards compatibility by passing through an options array to the end of the function